docs/en/contributing-to-appium/appium-packages.md

```diff
@@ -0,0 +1,255 @@
+- contains error classes for all types of errors
+  - sanitises error responses
+- (un)wraps params to commands
+- checks required params, validates params
+
+## appium-base-driver
+- designed to have a single testing session per instantiation
+- contains constraints on caps (platformName has to be present, etc)
+- validates capabilities
+- runs chain of promised commands with single concurrency
+- handles session restart
+- handles swipe options
+- exports class (DeviceSettings) to manage device settings (get/update)
+- contains basic commands
+  - to find elements
+  - create/delete sessions
+  - handle timeouts
+  - set/update device settings
+- provides helper methods for commands
+
+## appium-ios-driver
+- can also run as standalone server (has a small server part that extends from appium-express)
+- supported strategies: "name", "xpath", "id", "-ios uiautomation", "class name", "accessibility id”
+- can start simulator || real device (if udid is given)
+  - runs for each type a huge set of instructions
+    - removeInstrumentsSocket
+    - setBundleIdFromApp
+    - createInstruments
+    - runSimReset, isolateSimDevice, setLocale || runRealDeviceReset
+    - setPreferences || runRealDeviceReset
+    - startLogCapture
+    - prelaunchSimulator || (noop)
+    - startInstruments
+    - onInstrumentsLaunch
+    - configureBootstrap
+    - setBundleId
+    - setInitialOrientation
+    - initAutoWebview
+    - waitForAppLaunched
+  - all instructions are helper methods within driver.js
+- contains a more specific set of capability constraints
+- has logic to build and run Safari (safari-launcher) using appium-xcode
+- implements commands for iOS driver
+  - Selenium commands are compiled to ui-automator commands
+  - commands will be send out to uiAutoClient (appium-uiauto)
+- connects to appium remote debugger
+
+## appium-xcode
+- runs shell commands to return useful data from Xcode like
+  - getPath
+  - getVersion
+  - getAutomationTraceTemplatePath
+  - getMaxIOSSDK
+  - …
+- has an auto-retry built in
+
+## appium-uiauto
+- wrapper for the ios-automation framework
+- talks to it via socket connection
+- runs a command queue that get filled up by the sendCommand function
+- handles responses (as a buffer) from the ui-automation framework
+- uses osascript to rotate screenshots
+- provides method to bootstrap simulator/realdevice (dynamic-bootstrap)
+  - javascript files which are run in the uiautomation context (not node)
+  - responsible to execute actual ui automator commands
+  - see [UIAutomation docs](https://developer.apple.com/library/ios/documentation/DeveloperTools/Reference/UIAutomationRef/)
+  - command flow is like
+    - Webdriver command -> iOS driver command -> uiauto-command
+
+## appium-instruments
+- wrapper to run instruments commands
+- a lot of exec calls to talk to instruments binary
+- all of them take callbacks to propagate the result
+- contains also older versions of the instruments app (v4 - v7)
```

Comment:
Not sure what you mean by this. Are you meaning the "iwd" versions? This is "Instruments without delay", a fixing of Instruments to get rid of the delay between commands.